gptel-curl.el

```diff
@@ -66,6 +66,7 @@ PROMPTS is the data to send, TOKEN is a unique identifier."
       (gptel--log data "request body"))
     (append
      gptel-curl--common-args
+     (gptel-backend-curl-args gptel-backend)
```

Comment:
We could also use the backend OR the gptel-curl--common-args. That way the backend has full control over the options, but also the liability to add all required options that are in gptel-curl--common-args. Wdyt?

Reply:
Most of the arguments in `gtepl-curl--common-args` are critical for the functioning  of the callbacks/filter functions gptel uses. I think it's fine to leave these as "mandatory".